src/inmanta/config.py

```diff
@@ -115,35 +118,41 @@ def get(cls) -> ConfigParser:
 
     @overload
     @classmethod
-    def get(cls, section: str, name: str, default_value: Optional[str] = None) -> Optional[str]:
+    def get(cls, section: str, name: str, default_value: Optional[T] = None) -> Optional[str | T]:
         ...
 
     # noinspection PyNoneFunctionAssignment
     @classmethod
     def get(
-        cls, section: Optional[str] = None, name: Optional[str] = None, default_value: Optional[str] = None
-    ) -> Union[str, ConfigParser]:
+        cls, section: Optional[str] = None, name: Optional[str] = None, default_value: Optional[object] = None
+    ) -> object:
         """
         Get the entire config or get a value directly
         """
-        cfg = cls._get_instance()
         if section is None:
-            return cfg
+            return cls._get_instance()
 
         assert name is not None
         name = _normalize_name(name)
 
-        opt = cls.validate_option_request(section, name, default_value)
+        option: Optional[Option[object]] = cls.validate_option_request(section, name, default_value)
+        return cls.get_for_option(option) if option is not None else cls._get_value(section, name, default_value)
+
+    @classmethod
+    def get_for_option(cls, option: "Option[T]") -> T:
+        raw_value: Optional[str | T] = cls._get_value(option.section, option.name, option.get_default_value())
+        return option.validate(raw_value)
 
-        val = _get_from_env(section, name)
+    @classmethod
+    def _get_value(cls, section: str, name: str, default_value: Optional[T] = None) -> Optional[str | T]:
+        cfg: ConfigParser = cls._get_instance()
+        val: Optional[str] = _get_from_env(section, name)
```

Comment:
I'm curious, can the else block below return anything else than `default_value` ? If I understand correctly, this block is reached when the option is not found in the config or in the environment
```
val = cfg.get(section, name, fallback=default_value)
```

Reply:
This is the line that searches in the config file. It is reached when there's nothing in the environment.